wandb/sdk/launch/agent/agent.py

```diff
@@ -233,9 +251,24 @@ def update_status(self, status: str) -> None:
 
     def finish_thread_id(self, thread_id: int) -> None:
         """Removes the job from our list for now."""
+        job_and_run_status = self._jobs[thread_id]
+        if not job_and_run_status.run_id or not job_and_run_status.project:
+            self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
+        else:
+            try:
+                run_info = self._api.get_run_info(
+                    self._entity, job_and_run_status.project, job_and_run_status.run_id
+                )
+                # sweep runs exist but have no info before they are started
+                if run_info is None:
+                    self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
+            # normal runs just have no info
+            except CommError:
+                self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
```

Comment:
```suggestion
            run_info = None
            try:
                run_info = self._api.get_run_info(
                    self._entity, job_and_run_status.project, job_and_run_status.run_id
                )
            except CommError:
                # normal runs have no info, sweep runs have run_info=None
                pass
                
           if not run_info:
                self.fail_run_queue_item(job_and_run_status.run_queue_item_id)
```

Reply:
I feel like having self.fail_run_queue_item in only one place is slightly better